exporter/logicmonitorexporter/logs_exporter.go

```diff
@@ -0,0 +1,142 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logicmonitorexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/logicmonitorexporter"
+
+import (
+	"context"
+
+	"github.com/logicmonitor/lm-data-sdk-go/api/logs"
+	"github.com/logicmonitor/lm-data-sdk-go/utils"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+type logExporter struct {
+	config          *Config
+	logger          *zap.Logger
+	logIngestClient *logs.LMLogIngest
+	settings        component.TelemetrySettings
+}
+
+const (
+	hostname         = "hostname"
+	hostnameProperty = "system.hostname"
+)
+
+// Create new logicmonitor logs exporter
+func newLogsExporter(cfg component.Config, set component.ExporterCreateSettings) *logExporter {
+	oCfg := cfg.(*Config)
+
+	return &logExporter{
+		config:   oCfg,
+		logger:   set.Logger,
+		settings: set.TelemetrySettings,
+	}
+}
+
+func (e *logExporter) start(_ context.Context, host component.Host) error {
+	client, err := e.config.HTTPClientSettings.ToClient(host, e.settings)
+	if err != nil {
+		e.logger.Error("Failed to construct HTTP client", zap.Error(err))
+		return err
+	}
+	authParams := utils.AuthParams{
+		AccessID:    e.config.APIToken.AccessID,
+		AccessKey:   e.config.APIToken.AccessKey,
+		BearerToken: e.config.Headers["Authorization"],
+	}
+	options := []logs.Option{
+		logs.WithLogBatchingDisabled(),
+		logs.WithAuthentication(authParams),
+		logs.WithHTTPClient(client),
+		logs.WithEndpoint(e.config.Endpoint),
+	}
+
+	lli, err := logs.NewLMLogIngest(context.Background(), options...)
+	if err != nil {
+		e.logger.Debug("error in initializing LMLogIngest", zap.Error(err))
+		return err
+	}
+	e.logIngestClient = lli
+	return nil
+}
+
+func (e *logExporter) PushLogData(ctx context.Context, lg plog.Logs) (er error) {
```

Comment:
I doesn't look like your named return isn't being used, I would suggest removing it so it doesn't cause any issues.